sound/soc/sof/ipc4-priv.h

```diff
@@ -30,12 +30,14 @@ enum sof_ipc4_mtrace_type {
  * @sof_man4_module: Module info
  * @m_ida: Module instance identifier
  * @bss_size: Module object size
+ * @ext_library: Flag to indicate if module is from external library
  * @private: Module private data
  */
 struct sof_ipc4_fw_module {
 	struct sof_man4_module man4_module_entry;
 	struct ida m_ida;
 	u32 bss_size;
+	bool ext_library;
```

Comment:
@aiChaoSONG I'd like to see how this flag is really used before we merge this PR.

Reply:
I added two more commits to show the usage of the ext_library flag, and the base module config extension.

the base module config extension structure now is used to validate the size of kcontrol bytes, later it will also be be used to extract widget pin formats, similar to what I did here: https://github.com/thesofproject/linux/pull/4096/commits/016e755e034b17a1ba3bd2b4c9e921781125e5a2

@ujfalusi @ranj063 do you recommend the commit https://github.com/thesofproject/linux/pull/4121/commits/f019655964ba20424530d22f6c87ebcb92bd6e64 to be a fixup for  https://github.com/thesofproject/linux/pull/3961/commits/31b38da366b03b84d726e3f2ddd1464786324cbc?

Reply:
@aiChaoSONG Im not sure we really need this flag at all. All modules in the base FW should have the lib_id of 0 isnt it? And anything that's external will have lib_id > 0. Why not just use that?

Reply:
@ujfalusi I takes one of the suggestion from peter, see comments, https://github.com/thesofproject/linux/pull/4121#discussion_r1061213056

is a inline funtion testing on lib_id like this good for you?
```
static inline bool sof_ipc4_fw_module_is_in_basefw(struct sof_ipc4_fw_module *fw_module)
{
    u32 lib_id= fw_module->man4_module_entry.id & GENMASK(15, 12);
    return !lib_id;
}
```

Reply:
Yes that's perfect!